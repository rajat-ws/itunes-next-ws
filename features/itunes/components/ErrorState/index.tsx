/**
 *
 * ErrorState
 *
 */

import React from "react";
import get from "lodash/get";
import { compose } from "redux";
import { injectIntl, IntlShape } from "react-intl";
import { T, CustomCard } from "@common";
import { TrackResponse } from "../../api/getTracks";

interface ErrorStateProps {
  intl: IntlShape;
  loading: boolean;
  tracksData: TrackResponse | undefined;
  tracksError: string | undefined;
}

const ErrorState: React.FC<ErrorStateProps> = ({ intl, tracksData, tracksError, loading }) => {
  let trackError: string | undefined;
  if (tracksError) {
    trackError = tracksError;
  } else if (!get(tracksData, "resultCount", 0)) {
    trackError = "";
  }

  return !loading && trackError ? (
    <CustomCard
      color={tracksError ? "red" : "grey"}
      title={intl.formatMessage({ id: "something_went_wrong" })}
      data-testid="error-state"
    >
      <T id={trackError} />
    </CustomCard>
  ) : null;
};

export default compose(injectIntl)(ErrorState);
