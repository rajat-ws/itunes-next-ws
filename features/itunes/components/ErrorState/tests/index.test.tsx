import { render } from "@app/utils/testUtils";
import ErrorState from "../index";

describe("<ErrorState />", () => {
  const errorStateProps = {
    intl: {},
    tracksData: undefined,
    tracksError: "error",
    loading: false,
  };

  it("should render and match the snapshot", () => {
    const { baseElement } = render(<ErrorState {...errorStateProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it("should contain 1 ErrorState component", () => {
    const { getByTestId } = render(<ErrorState {...errorStateProps} />);
    expect(getByTestId("error-state")).toBeInTheDocument();
  });
});
