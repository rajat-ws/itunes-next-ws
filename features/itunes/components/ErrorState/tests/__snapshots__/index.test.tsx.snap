// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorState /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-card ant-card-bordered styled__CustomCard-sc-15gkald-1 hqWzjF"
      color="red"
      data-testid="error-state"
    >
      <div
        class="ant-card-head"
      >
        <div
          class="ant-card-head-wrapper"
        >
          <div
            class="ant-card-head-title"
          >
            Sorry. Something went wrong! Please try again in sometime.
          </div>
        </div>
      </div>
      <div
        class="ant-card-body"
      >
        <p
          class="T__StyledText-sc-apdcd0-0 fjJkvM"
          data-testid="t"
        >
          error
        </p>
      </div>
    </div>
  </div>
</body>
`;
